server/controllers/payment.js

```diff
@@ -0,0 +1,139 @@
+const asyncHandler = require("express-async-handler");
+const Payment = require("../models/Payment");
+const stripe = require("stripe")(process.env.STRIPE_SECRET_KEY);
+
+// @route GET /payments
+// @desc Get all payment from user
+// @access Private
+exports.getAllPayments = asyncHandler(async (req, res, next) => {
+  const payment = await Payment.find({ userId: req.user.id });
+  if (payment) {
+    res.status(200).json({
+      success: {
+        payment: payment,
+      },
+    });
+  } else {
+    res.status(404);
+    throw new Error("Payments Not Found");
+  }
+});
+
+// @route GET /payments/:id
+// @desc Get certain payment from id
+// @access Private
+exports.getPayment = asyncHandler(async (req, res, next) => {
+  const { id } = req.params;
+  const payment = await Payment.findById(id);
+  if (payment) {
+    res.status(200).json({
+      success: {
+        payment: payment,
+      },
+    });
+  } else {
+    res.status(404);
+    throw new Error("Pls enter valid payment id");
+  }
+});
+
+// @route PUT /payments/:id/pay
+// @desc Pay current unpaid payment
+// @access Private
+exports.makePayment = asyncHandler(async (req, res, next) => {
+  const { id } = req.params;
+  const payment = await Payment.findOne({ id });
+
+  if (!payment) {
+    res.status(404);
+    throw new Error("Payment doesn't exist");
+  }
+
+  if (payment.isPaid) {
+    res.status(400);
+    throw new Error("This payment has already be made");
+  }
+
+  const paymentConfirmation = await stripe.paymentIntents.confirm(payment.paymentIntentId, {
+    payment_method: "pm_card_visa",
```

Comment:
@cl-codes When working on your payment methods task, we should make sure to update this.